Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,130 @@
+
+import ArgumentParser
+import CoreCommands
+import TSCUtility
+import TSCBasic
+import PackageModel
+import Foundation
+
+extension SwiftPackageTool {
+    
+    struct Install: SwiftCommand {
+        static var configuration: CommandConfiguration {
+            CommandConfiguration(abstract: "Offers the ability to install executable targets of the current package.")
+        }
+        
+        @OptionGroup()
+        var globalOptions: GlobalOptions
+        
+        func run(_ tool: SwiftTool) throws {
+            let dotSwiftPMDir = try tool.fileSystem.getOrCreateDotSwiftPMDirectory()
+            
+            let env = ProcessInfo.processInfo.environment
+            
+
+            // ~/.swiftpm/bin/
+            let swiftpmBinDir = dotSwiftPMDir.appending(component: "bin")
+            
+            if let path = env.path, !path.contains(swiftpmBinDir.pathString), env["SWIFTPM_SHUT_UP_ABOUT_EXECUTABLE_PATH"] == nil {
+                tool.observabilityScope.emit(warning: "PATH doesn't include \(swiftpmBinDir.pathString)! This means you won't be able to access the installed executables by default, and will need to specify the full path. If you wish to stop seeing this warning, set the enviroment variable `SWIFTPM_SHUT_UP_ABOUT_EXECUTABLE_PATH`.")
+            }
+            
+            let registryURL = dotSwiftPMDir.appending(component: "installedPackages.json").asURL
+            
+            var alreadyExisting = (try? InstalledPackage.registered(registryURL: registryURL)) ?? []
+            
+            let workspace = try tool.getActiveWorkspace()
+            let packageRoot = try tool.getPackageRoot()
+
+            let packageGraph = try tsc_await {
+                workspace.loadRootPackage(at: packageRoot, observabilityScope: tool.observabilityScope, completion: $0)
+            }
+            
+            let possibleCanidates = packageGraph.products.filter { $0.type == .executable }
+            let productToInstall: Product
+            
+            switch possibleCanidates.count {
+            case 0:
+                throw StringError("No Executable Products in Package.swift.")
+            case 1:
+                productToInstall = possibleCanidates[0]
+            default: // More than one, ask the user which one they wanna install
+                print("More than one executable target selected, please select of the following which you'd like to install by typing the number and pressing enter:")
+                for (index, product) in possibleCanidates.enumerated() {
+                    print("[\(index)] \(product.name)")
+                }
+                
+                guard let input = readLine(), let int = Int(input), possibleCanidates.indices.contains(int) else {
+                    throw StringError("Input should be a number between 0 and \(possibleCanidates.count - 1).")
+                }
+                
+                productToInstall = possibleCanidates[int]
+            }
```

Comment:
It's not conventional for SwiftPM to require interactive user input, especially if users may run these commands on CI, which is necessarily non-interactive. This command should take a `--product` option that disambiguates the product to install. IMO if `--product` is not specified it should install of the top-level package products.